lib/iris/tests/unit/cube/test_Cube.py

```diff
@@ -549,6 +550,93 @@ def test_single_string_aggregation(self):
                          AuxCoord(['a|a', 'a'], long_name='bar'))
 
 
+class Test_aggregated_by__lazy(tests.IrisTest):
+    def setUp(self):
+        self.data = np.arange(44).reshape(4, 11)
+        self.lazydata = as_lazy_data(self.data)
+        self.cube = Cube(self.lazydata)
+
+        val_coord = AuxCoord([0, 0, 0, 1, 1, 2, 0, 0, 2, 0, 1],
+                             long_name="val")
+        label_coord = AuxCoord(['alpha', 'alpha', 'beta',
+                                'beta', 'alpha', 'gamma',
+                                'alpha', 'alpha', 'alpha',
+                                'gamma', 'beta'],
+                               long_name='label', units='no_unit')
+        simple_agg_coord = AuxCoord([1, 1, 2, 2], long_name='simple_agg')
+        spanning_coord = AuxCoord(np.arange(44).reshape(4, 11),
+                                  long_name='spanning')
+        spanning_label_coord = AuxCoord(
+            np.arange(1, 441, 10).reshape(4, 11).astype(str),
+            long_name='span_label', units='no_unit')
```

Comment:
The spanning and span_label 2D coordinates are added to the cube but there isn't any testing of whether they are handled correctly during aggregation.
You could add an extra check that their points (and bounds for the numerical coordinate) are as expected.

Reply:
Thanks @lbdreyer! I've been mulling this over a little bit and I think I'd rather remove the spanning stuff from the setup. The reason is that this part tests the part of *what* to aggregate, whereas this pr only changes some part of *how* to perform it. The alternative would be a verbatim copy of the non lazy test, but I think it really wouldn't add much. Would that be ok?

Reply:
Hi @zklaus sorry for delay.
I think it's quite good enough to just remove the spanning coords, and then I think we can merge this as it is.
**Can you manage that within about the next 2 days, so it can go into Iris 2.3 ?**
Thanks.

Reply:
Yes